front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -252,152 +252,163 @@ export const usePathfinding = (
+        );
+
+        const theoreticalMargin = i === 0 ? step.theoreticalMargin || '0%' : step.theoreticalMargin;
+
+        const stopFor = i === pathSteps.length - 1 && !step.stopFor ? '0' : step.stopFor;
+        const stopType = i === pathSteps.length - 1 && !step.stopFor ? undefined : step.stopType;
+
+        return {
+          ...step,
+          positionOnPath: pathResult.path_item_positions[i],
+          stopFor,
+          stopType,
+          theoreticalMargin,
+          ...(correspondingOp && {
+            name: correspondingOp.name,
+            uic: correspondingOp.uic,
+            ch: correspondingOp.ch,
+            kp: correspondingOp.kp,
+            coordinates: correspondingOp.coordinates,
+          }),
+        };
+      });
+
+      if (!isEmptyArray(powerRestrictions)) {
+        dispatch(
+          setWarning({
+            title: t('warningMessages.pathfindingChange'),
+            text: t('warningMessages.powerRestrictionsReset'),
+          })
+        );
+      }
+      dispatch(updatePathSteps(updatedPathSteps));
+
+      const allWaypoints = upsertPathStepsInOPs(
+        suggestedOperationalPoints,
+        compact(updatedPathSteps)
+      );
+
+      setPathProperties?.({
+        electrifications,
+        geometry,
+        suggestedOperationalPoints,
+        allWaypoints,
+        length: pathResult.length,
+        trackSectionRanges: pathResult.track_section_ranges,
+        incompatibleConstraints,
+      });
+    };
+
     const startPathFinding = async () => {
-      if (!pathfindingState.running) {
-        pathfindingDispatch({ type: 'PATHFINDING_STARTED' });
-        const pathfindingInput = generatePathfindingParams();
-        if (!pathfindingInput || !infraId) {
-          dispatch(
-            setFailure({
-              name: t('pathfinding'),
-              message: t('pathfindingMissingParamsSimple'),
-            })
+      if (pathfindingState.running) {
+        return;
+      }
+
+      pathfindingDispatch({ type: 'PATHFINDING_STARTED' });
+      const pathfindingInput = generatePathfindingParams();
+      if (!pathfindingInput) {
+        dispatch(
+          setFailure({
+            name: t('pathfinding'),
+            message: t('pathfindingMissingParamsSimple'),
+          })
+        );
+        return;
+      }
+
+      try {
+        const pathfindingResult = await postPathfindingBlocks(pathfindingInput).unwrap();
+
+        if (pathfindingResult.status === 'success') {
+          await populateStoreWithSuccessfulPathfinding(pathfindingResult);
+          pathfindingDispatch({ type: 'PATHFINDING_FINISHED' });
+          return;
+        }
+
+        const incompatibleConstraintsCheck =
+          pathfindingResult.failed_status === 'pathfinding_not_found' &&
+          pathfindingResult.error_type === 'incompatible_constraints';
+
+        if (incompatibleConstraintsCheck) {
+          await populateStoreWithSuccessfulPathfinding(
```

Comment:
it’s a bit puzzling to have this function named `withSuccessfulPathfinding` in a case where the status is actually failed.